File: lib/components/rqm_workspace_component.dart
import 'package:angular/angular.dart';
import 'package:openrqm_client_desktop_nwjs/components/rqm_document_component.dart';
import 'package:openrqm/api.dart';

@Component(
  selector: 'rqm-workspace',
  template: '''
    <div classes="workspaceTreeWorkspace" *ngIf="name != null && name != ''">{{name}}</div>
      <ul *ngIf="workspaces != null || documents != null">
        <div *ngIf="workspaces != null">
          <li *ngFor="let workspace of workspaces" >
              <rqm-workspace [name]="workspace.name" [workspaceId]="workspace.workspaceId" [workspaces]="workspace.workspaces" [documents]="workspace.documents"></rqm-workspace>  
          </li>
        </div>
        <div *ngIf="documents != null">
          <li *ngFor="let document of documents">
              <rqm-document [name]="document.name">
              </rqm-document>
          </li>
        </div>
      </ul>
  ''',
  directives: [
    coreDirectives,
    RQMWorkspaceComponent,
    RQMDocumentComponent,
  ],
)
class RQMWorkspaceComponent {
  @Input()
  String name;
  @Input()
  int workspaceId;
  @Input()
  List<RQMWorkspace> workspaces;
  @Input()
  List<RQMDocument> documents;

  RQMWorkspaceComponent({
    this.name,
    this.workspaceId,
    this.workspaces,
    this.documents,
  });
}

File: src/app/rqmworkspace/rqmworkspace.component.ts
import { Component, OnInit, Input } from '@angular/core';
import { RQMWorkspace, RQMDocument } from 'openrqm-api';

@Component({
  selector: 'rqm-workspace',
  templateUrl: './rqmworkspace.component.html',
  styleUrls: ['./rqmworkspace.component.scss']
})
export class RQMWorkspaceComponent implements OnInit {
  @Input() name: string;
  @Input() workspaceId: number;
  @Input() workspaces: RQMWorkspace[];
  @Input() documents: RQMDocument[];

  constructor() {

  }

  ngOnInit() {
  }

}
